iib/workers/tasks/build.py

```diff
@@ -816,7 +816,10 @@ def handle_add_request(
             arch = sorted(arches)[0]
             log.info('Building a temporary index image to satisfy the deprecation requirement')
             _build_image(temp_dir, 'index.Dockerfile', request_id, arch)
-            intermediate_image_name = _get_local_pull_spec(request_id, arch, include_transport=True)
+            _push_image(request_id, arch)
```

Comment:
Thanks for the note of `merge-index-image`, it uses the same logic.

Now I've 2 questions:
1. Do we have a test for it somewhere to make sure it works?
2. Do we need or have a way to clean-up the registry to not let it store the intermediate images forever?

Reply:
1. iib-api-test should cover this on any instance (but I guess they are failing right now) - maybe because of this issue. 
2. no - it should be overwritten later by push to the same pull spec